### ISSUE TITLE
Image size (324768098 pixels) exceeds limit of 178956970 pixels, could be decompression bomb DOS attack.

### ISSUE THREAD

**Issue by safal-305** [ultralytics/yolov5#7003]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

WARNING: C:\Users\K\yolov5\data\images\blur\images\train\GCC9.jpeg: ignoring corrupt image/label: Image size (324768098 pixels) exceeds limit of 178956970 pixels, could be decompression bomb DOS attack.

### Environment

yolov5
windows10
python3.9

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@safal-305 your jpeg has problems that are unrelated to YOLOv5. Before training make sure that your jpeg is readable with cv2.imread('im.jpg')

**Comment by georgegoglodze** [comment]
Hello, I have the same issue. My image files are large can this be a problem?
PS I can read files with cv2
>> cv2.imread('cam5.png')
array([[[1, 1, 1],
        [0, 0, 0],
        [0, 0, 0],
        ...,
        [0, 0, 0],
        [0, 0, 0],
        [0, 0, 0]],

**Comment by Adesoji1** [comment]
The `DecompressionBombError` you're encountering indicates that you are trying to load an image that has an extremely large number of pixels, which Pillow (the library used by `tensorflow.keras.preprocessing.image.load_img` under the hood) identifies as a potential risk for a decompression bomb DOS attack. This is a security feature to protect your system against such attacks that use image files to exploit vulnerabilities.

To handle this, you can do a couple of things:

1. Increase the maximum allowed pixel limit in Pillow.
2. Check the image's validity before processing it.

Here's how you can apply both approaches:

Firstly, to increase the pixel limit in Pillow (which is used by Keras/TensorFlow internally), you can add the following lines before you load your images:

```python
from PIL import Image

# Increase the pixel limit
Image.MAX_IMAGE_PIXELS = None  # or set to a larger value like 933120000
```

Secondly, you can use OpenCV to verify the images before passing them to your TensorFlow model. You would need to install OpenCV if you haven't already. You can do so using pip:

```bash
pip install opencv-python-headless
```

And then you can use OpenCV to read the images:

```python
import cv2

def is_valid_image(file_path):
    """ Check if the image is valid. """
    img = cv2.imread(file_path)
    return img is not None

# ... Inside your loop:

for img_file in image_files:
    img_path = os.path.join(images_path, img_file)
    if is_valid_image(img_path):
        features_dict_vgg16[img_file] = extract_features_vgg16(img_path)
    else:
        print(f"Invalid image file detected: {img_file}")
```

By setting `Image.MAX_IMAGE_PIXELS` to `None`, you remove the limitation on image size, but be aware that this might expose you to the risk of a decompression bomb if you're dealing with untrusted image sources.

By checking whether `cv2.imread` returns `None`, you're ensuring that you only try to process valid images, which can help avoid issues when trying to open or process corrupted or invalid image files.

Please handle these settings with care, especially if you're working with images from an untrusted source.

**Comment by glenn-jocher** [comment]
@Adesoji1 the `DecompressionBombError` you encountered is due to an image with an extremely large number of pixels, which Pillow (used by `tensorflow.keras.preprocessing.image.load_img`) identifies as a potential risk for a decompression bomb DOS attack. This is a security feature implemented to protect against such attacks that exploit vulnerabilities using image files.

You can address this issue by following these steps:

1. Increase the maximum allowed pixel limit in Pillow. You can do this by adding the following lines before loading your images:
```python
from PIL import Image

# Increase the pixel limit
Image.MAX_IMAGE_PIXELS = None  # or set to a larger value like 933120000
```

2. Validate the images before processing them using OpenCV. If you haven't installed OpenCV yet, you can do so using pip:
```bash
pip install opencv-python-headless
```

Next, you can use OpenCV to verify the images before passing them to your TensorFlow model. Here's an example code snippet:

```python
import cv2

def is_valid_image(file_path):
    """ Check if the image is valid. """
    img = cv2.imread(file_path)
    return img is not None

# ... Inside your loop:

for img_file in image_files:
    img_path = os.path.join(images_path, img_file)
    if is_valid_image(img_path):
        features_dict_vgg16[img_file] = extract_features_vgg16(img_path)
    else:
        print(f"Invalid image file detected: {img_file}")
```

Setting `Image.MAX_IMAGE_PIXELS` to `None` removes the limitation on image size, but use this with caution, especially when dealing with untrusted image sources as it may expose you to the risk of a decompression bomb.

By checking whether `cv2.imread` returns `None`, you can ensure that only valid images are processed, helping to avoid issues with corrupted or invalid image files.

Please handle these settings carefully, especially when working with images from untrusted sources.
